src/cs/Ssh/Services/AuthenticationService.cs

```diff
@@ -171,17 +199,62 @@ await HandleMessageAsync(message.ConvertTo<PasswordRequestMessage>(), cancellati
 		// of the client's username and password.
 		var args = new SshAuthenticatingEventArgs(
 			SshAuthenticationType.ClientPassword,
-			username: message.Username!,
+			username: message.Username,
 			password: message.Password ?? string.Empty);
-		await HandleAuthenticatingAsync(message, args, cancellation)
-			.ConfigureAwait(false);
+		await HandleAuthenticatingAsync(args, cancellation).ConfigureAwait(false);
+	}
+
+	private async Task BeginInteractiveAuthenticationAsync(
+		AuthenticationRequestMessage message, CancellationToken cancellation)
+	{
+		// Raise an Authenticating event that allows the server to interactively prompt for
+		// information from the client.
+		var args = new SshAuthenticatingEventArgs(
+			SshAuthenticationType.ClientInteractive,
+			username: message.Username,
+			infoRequest: null,
+			infoResponse: null);
+		await HandleAuthenticatingAsync(args, cancellation).ConfigureAwait(false);
+	}
+
+	private async Task HandleMessageAsync(
+		AuthenticationInfoRequestMessage message, CancellationToken cancellation)
+	{
+		// Raise an Authenticating event that allows the client to respond to interactive prompts
+		// and provide requested information to the server.
+		var args = new SshAuthenticatingEventArgs(
+			SshAuthenticationType.ClientInteractive,
+			username: null,
+			infoRequest: message,
+			infoResponse: null);
+		await HandleAuthenticatingAsync(args, cancellation).ConfigureAwait(false);
+	}
+
+	private async Task HandleMessageAsync(
+		AuthenticationInfoResponseMessage message, CancellationToken cancellation)
+	{
+		// Raise an Authenticating event that allows the server to process the client's responses
```

Comment:
How will this behave if `Authenticating` event handler is not set? Will it just hang never responding back?

Reply:
I see that it would fail auth, good.